source/state_representation/include/state_representation/trajectory/CartesianTrajectory.hpp

```diff
@@ -0,0 +1,165 @@
+#pragma once
+
+#include "state_representation/space/cartesian/CartesianState.hpp"
+#include "state_representation/trajectory/TrajectoryBase.hpp"
+
+namespace state_representation {
+class CartesianTrajectory : public TrajectoryBase<Eigen::VectorXd> {
+private:
+  std::string reference_frame_;///< name of the reference frame
+
+public:
+  /**
+   * @brief Constructor with name and reference frame provided
+   * @param reference_frame reference frame of the trajectory points
+   * @param name the name of the state
+   */
+  explicit CartesianTrajectory(const std::string& reference_frame = "world", const std::string& name = "");
```

Comment:
Name is usually first in control libraries. Also it might be neat to have a way of constructing an empty trajectory and then whenever the first point is added, the reference frame of that point is taken as reference frame for the whole trajectory. Not sure if that is possible though.

Additionally, I think a constructor that takes the first point is also reasonable.

Reply:
I added a 2 more constructors to that make use of the `set_point` and `set_points`. I think that should do it